node/web/public/scripts/controllers/authController.js

```diff
@@ -1,17 +1,84 @@
 Hilary.scope('heinz').register({
     name: 'authController',
-    dependencies: ['newGidgetModule', 'GidgetRoute', 'locale', 'viewEngine'],
-    factory: function ($this, GidgetRoute, locale, viewEngine) {
+    dependencies: ['newGidgetModule', 'GidgetRoute', 'locale', 'viewEngine', 'ProfileVM'],
+    factory: function ($this, GidgetRoute, locale, viewEngine, ProfileVM) {
         'use strict';
 
-        // GET /#/login
-        // login
+        $this.get['/checkLogin'] = new GidgetRoute({
+            routeHandler: function (err,req) {
+                $.ajax({
+                    url: '/logincheck'
+                }).done(function (data) {
+                    if (data === '200') {
+                        window.location.replace('/profile');
```

Comment:
by using this instead of the router, you reload the entire page, defeating the value of having a Single Page Application (SPA).

> note this occurs several times in your app, I won't comment on the others.